be/src/vec/exec/vrepeat_node.cpp

```diff
@@ -0,0 +1,244 @@
+        cur_col++;
+    }
+
+    // Fill grouping ID to tuple
+    for (auto slot_idx = 0; slot_idx < _grouping_list.size(); slot_idx++) {
+        DCHECK_LT(slot_idx, _virtual_tuple_desc->slots().size());
+        const SlotDescriptor* _virtual_slot_desc = _virtual_tuple_desc->slots()[slot_idx];
+        DCHECK_EQ(_virtual_slot_desc->type().type, _output_slots[cur_col]->type().type);
+        DCHECK_EQ(_virtual_slot_desc->col_name(), _output_slots[cur_col]->col_name());
+        int64_t val = _grouping_list[slot_idx][repeat_id_idx];
+        auto* column_ptr = columns[cur_col].get();
+        if (_output_slots[cur_col]->is_nullable()) {
+            auto* nullable_column = reinterpret_cast<ColumnNullable *>(columns[cur_col].get());
+            auto& null_map = nullable_column->get_null_map_data();
+            column_ptr = &nullable_column->get_nested_column();
+
+            for (size_t i = 0; i < child_block->rows(); ++i) {
+                null_map.push_back(0);
+            }
+        }
+
+        auto* col = assert_cast<ColumnVector<Int64> *>(column_ptr);
+        for (size_t i = 0; i < child_block->rows(); ++i) {
+            col->insert_value(val);
+        }
+        cur_col ++;
+    }
+
+    DCHECK_EQ(cur_col, column_size);
+
+    if (!columns.empty() && !columns[0]->empty()) {
+        auto n_columns = 0;
+        if (!mem_reuse) {
+            for (const auto slot_desc : _output_slots) {
+                output_Block->insert(
+                    ColumnWithTypeAndName(std::move(columns[n_columns++]), slot_desc->get_data_type_ptr(), slot_desc->col_name()));
+            }
+        } else {
+            columns.clear();
+        }
+    }
+    return Status::OK();
+}
+
+Status VRepeatNode::get_next(RuntimeState* state, Block* block, bool* eos) {
+    VLOG_CRITICAL << "VRepeatNode::get_next";
+    SCOPED_TIMER(_runtime_profile->total_time_counter());
+
+    if (state == NULL || block == NULL || eos == NULL)
+        return Status::InternalError("input is NULL pointer");
+
+    RETURN_IF_ERROR(exec_debug_action(TExecNodePhase::GETNEXT));
+    RETURN_IF_CANCELLED(state);
+    DCHECK(_repeat_id_idx >= 0);
+    for (const std::vector<int64_t>& v : _grouping_list) {
+        DCHECK(_repeat_id_idx <= (int)v.size());
+    }
+    DCHECK(block->rows() == 0);
+
+    // current child block has finished its repeat, get child's next block
+    if (_child_block.get() == nullptr) {
+        if (_child_eos) {
+            *eos = true;
+            return Status::OK();
+        }
+
+        _child_block.reset(new Block());
+        RETURN_IF_ERROR(child(0)->get_next(state, _child_block.get(), &_child_eos));
+
+        if (_child_block->rows() <= 0) {
+            _child_block.reset(nullptr);
+            *eos = true;
+            return Status::OK();
+        }
+    }
+
+    RETURN_IF_ERROR(get_repeated_block(_child_block.get(), _repeat_id_idx, block));
+
+    _repeat_id_idx++;
+
+    int size = _repeat_id_list.size();
+    if (_repeat_id_idx >= size) {
```

Comment:
why here reset the _child_block， call `release_block_memory` to reuse the mem of _child_block